source/common/network/io_socket_handle_impl.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include "envoy/network/io_handle.h"
+
+namespace Envoy {
+namespace Network {
+
+/**
+ * IoHandle derivative for sockets
+ */
+class IoSocketHandle : public IoHandle {
+public:
+  IoSocketHandle(int fd = -1) : fd_(fd) {}
+
+  ~IoSocketHandle() {}
+
+  /**
+   * TODO(sbelair2)  To be removed when the IoSocketHandle derivative is integrated
+   * and the fd is fully abstracted from clients.
+   *
+   * @return the socket's file descriptor in the handle.
+   */
```

Comment:
This comment can get moved to the base class

Generally for impls like this we just comment

// From IoHandle
int fd()....
void close....

and folks refer to the base class for docs.

Reply:
Got it.